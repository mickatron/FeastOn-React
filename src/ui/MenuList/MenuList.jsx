import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/shared/proptypes/className';
// import NavLink from '../NavLink/NavLink';

export default class MenuList extends React.PureComponent {
  
  static propTypes = {
    className: classNameType,
    items: PropTypes.arrayOf( 
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,        
        PropTypes.node,
      ]).isRequired,
    ),
    layout: PropTypes.oneOf(['block', 'inline']),
  };

  static defaultProps = {
    layout: 'block',
  }

  render() {
    const { className, items, layout } = this.props;
    const otherClassNames = {
      'c-MenuList--inline': layout === 'inline',
    };

    return( 
      <ul className={classnames(className, 'c-MenuList', otherClassNames)}>
        {items.map(function(menuItem, index) {
          return(
            <li 
              key={index}
            >
              { menuItem }
            </li>
          );
        })}
      </ul>
    );
  }
}