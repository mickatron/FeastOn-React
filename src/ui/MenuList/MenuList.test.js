
import { shallow } from 'enzyme';
import React from 'react';
import MenuList from './MenuList';

describe('MenuList', () => {

  let component = null;
  beforeAll(() => {
    component = shallow(
      <MenuList 
        className="test"
        items={[
          { label: 'item 1', route: '/route' }, 
          { label: 'item 1', route: '/route' },
          { label: 'Configure', route: '/route' },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders', () => {
    expect(component.find('.c-MenuList.test').exists()).toBeTruthy();
  });

  it('three menu items', () => {
    expect(component.find('.c-MenuList li')).toHaveLength(3);
  });


});