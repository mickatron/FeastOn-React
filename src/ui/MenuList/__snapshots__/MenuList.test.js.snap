// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ul
  className="test c-MenuList"
>
  <li
    key="0"
  />
  <li
    key="1"
  />
  <li
    key="2"
  />
</ul>
`;
